lib/faraday/response/logger.rb

```diff
@@ -4,78 +4,93 @@
 
 module Faraday
   class Response::Logger < Response::Middleware
-    extend Forwardable
-
-    DEFAULT_OPTIONS = { headers: true, bodies: false }
-
     def initialize(app, logger = nil, options = {})
       super(app)
-      @logger = logger || begin
-        require 'logger'
-        ::Logger.new($stdout)
-      end
-      @filter = []
-      @options = DEFAULT_OPTIONS.merge(options)
-      yield self if block_given?
+      formatter_class = options.delete(:formatter) || Formatter
+      @formatter = formatter_class.new(logger: logger, options: options)
+      yield @formatter if block_given?
     end
 
-    def_delegators :@logger, :debug, :info, :warn, :error, :fatal
-
     def call(env)
-      info('request')  { "#{env.method.upcase} #{apply_filters(env.url.to_s)}" }
-      debug('request') { apply_filters(dump_headers(env.request_headers)) } if log_headers?(:request)
-      debug('request') { apply_filters(dump_body(env[:body])) } if env[:body] && log_body?(:request)
+      @formatter.request(env)
       super
     end
 
     def on_complete(env)
-      info('response')  { "Status #{env.status}" }
-      debug('response') { apply_filters(dump_headers(env.response_headers)) } if log_headers?(:response)
-      debug('response') { apply_filters(dump_body(env[:body])) } if env[:body] && log_body?(:response)
+      @formatter.response(env)
     end
 
-    def filter(filter_word, filter_replacement)
-      @filter.push([filter_word, filter_replacement])
-    end
+    class Formatter
```

Comment:
Although this class is directly linked to this logger, we're trying to isolate all classes into separate files do make the codebase a little cleaner.
@technoweenie what do you think of moving the formatter on a separate file?

Reply:
Lets move it to `faraday/logging/formatter.rb` this was it wouldn't have a misleading `Request` namespace. I'll add a commit for that.